### PR TITLE
Add support for multiple extension overrides

### PR DESCRIPTION
**Pull request by maxandersen**

### CODE REVIEW

**Review by aloubyansky — APPROVED** [review]

**Review by quintesse — APPROVED** [review]
LGTM

File: devtools/platform-descriptor-json-plugin/src/main/java/io/quarkus/maven/GenerateExtensionsJsonMojo.java [hunk 1]
```diff
@@ -9,20 +9,10 @@
...
-import javax.json.Json;
-import javax.json.JsonArray;
-import javax.json.JsonArrayBuilder;
-import javax.json.JsonObject;
-import javax.json.JsonObjectBuilder;
-import javax.json.JsonReader;
-import javax.json.JsonValue;
-import javax.json.JsonWriter;
-import javax.json.JsonWriterFactory;
+import java.util.*;
```
Comment (gastaldi) [inline devtools/platform-descriptor-json-plugin/src/main/java/io/quarkus/maven/GenerateExtensionsJsonMojo.java]: We don't use star imports, can you use the appropriate code formatter?

File: devtools/platform-descriptor-json-plugin/src/main/java/io/quarkus/maven/GenerateExtensionsJsonMojo.java [hunk 2]
```diff
@@ -121,29 +112,25 @@ public void execute() throws MojoExecutionException, MojoFailureException {
...
+        for (File file : overrides) {
+            OverrideInfo overrideInfo = getOverrideInfo(file);
+            if (overrideInfo != null) {
+                allOverrides.add(overrideInfo);
             }
         }
 
+        //Map<String, JsonObject> extOverrides = overrideInfo.getExtOverrides();
+        //JsonObject theRest = overrideInfo.getTheRest();
+
```
Comment (gastaldi) [inline devtools/platform-descriptor-json-plugin/src/main/java/io/quarkus/maven/GenerateExtensionsJsonMojo.java]: Remove commented out code

File: devtools/platform-descriptor-json-plugin/src/main/java/io/quarkus/maven/GenerateExtensionsJsonMojo.java [hunk 3]
```diff
@@ -121,29 +112,25 @@ public void execute() throws MojoExecutionException, MojoFailureException {
...
+        List<File> overrides = new ArrayList<>();
 
-                theRest = overridesObject;
-            } catch (IOException e) {
-                throw new MojoExecutionException("Failed to read " + overridesFile, e);
+        overrides.addAll(Arrays.stream(overridesFile.split(","))
+                .map(s -> {
+                    return new File(s.trim());
+                })
+                .collect(Collectors.toList()));
```
Comment (gastaldi) [inline devtools/platform-descriptor-json-plugin/src/main/java/io/quarkus/maven/GenerateExtensionsJsonMojo.java]:
```suggestion
        for (String file: overridesFile.split(",")) {
           overrides.add(new File(file.trim()));
        }
```

File: devtools/platform-descriptor-json-plugin/src/main/java/io/quarkus/maven/GenerateExtensionsJsonMojo.java [hunk 4]
```diff
@@ -121,29 +112,25 @@ public void execute() throws MojoExecutionException, MojoFailureException {
...
+
+        List<OverrideInfo> allOverrides = new ArrayList();
+        for (File file : overrides) {
```
Comment (gastaldi) [inline devtools/platform-descriptor-json-plugin/src/main/java/io/quarkus/maven/GenerateExtensionsJsonMojo.java]: You can iterate `overridesFile.split(",")` and delete the `List<File> overrides` above :)

**Review by gastaldi — CHANGES_REQUESTED** [review]
From the added comments, the code format one is the most important I think :)

File: devtools/platform-descriptor-json-plugin/src/main/java/io/quarkus/maven/GenerateExtensionsJsonMojo.java (hunk 1 above)
Comment (maxandersen) [reply devtools/platform-descriptor-json-plugin/src/main/java/io/quarkus/maven/GenerateExtensionsJsonMojo.java]: the build passes so I assume I am already using the appropriate code formatter.

Comment (maxandersen) [reply devtools/platform-descriptor-json-plugin/src/main/java/io/quarkus/maven/GenerateExtensionsJsonMojo.java]: oh damn - is the PR's i made on having devtools share the formatting still not in master. ? gddmmit

File: devtools/platform-descriptor-json-plugin/src/main/java/io/quarkus/maven/GenerateExtensionsJsonMojo.java (hunk 4 above)
Comment (maxandersen) [reply devtools/platform-descriptor-json-plugin/src/main/java/io/quarkus/maven/GenerateExtensionsJsonMojo.java]: true - good catch. it became this way as I was originally merging two lists but that’s not relevant anymore fixing.

File: devtools/platform-descriptor-json-plugin/src/main/java/io/quarkus/maven/GenerateExtensionsJsonMojo.java (hunk 1 above)
Comment (maxandersen) [reply devtools/platform-descriptor-json-plugin/src/main/java/io/quarkus/maven/GenerateExtensionsJsonMojo.java]: fixed manually

**Comment by maxandersen** [comment]
> From the added comments, the code format one is the most important I think :)

pushed what should fix your various comments.

**Review by gastaldi — APPROVED** [review]
LGTM, thanks!

**Comment by gastaldi** [comment]
Oops, CI failed

**Comment by maxandersen** [comment]
lets try that again ;)

**Pull request merged**